File: addresstocoordinates.py
import json
import urllib
import time
import csv
import pprint

apikey = "your google API key here"
filename = "cities.csv"
sep = ";"

def query(addr):
    try:
        req = urllib.request.urlopen("https://maps.googleapis.com/maps/api/geocode/json?address=" + addr + "&key=" + apikey)
    except err:
        print(err)
    j = json.loads(req.read())
    if j['status'] != "OK":
        raise Exception('Google said ' + addr + ' query was not OK')
        
    return j['results']

resolved = {}

with open(filename) as fd:
    for addr in fd.readlines():
        for result in query(addr):
            l = result['geometry']['location']
            resolved[addr.strip()] = l
            time.sleep(1)

for key in resolved:
    print(key + sep + str(resolved[key]['lat']) + sep + str(resolved[key]['lng']))

File: addresstocoordinates.js
var apikey = "your google API key here"
var coords = [];
var fileuri = "cities.csv";
var container = document.getElementById("csvrows");

var resolve = function(addr, bar) {
	var xhr = new XMLHttpRequest();
	query = "https://maps.googleapis.com/maps/api/geocode/json?address=" + addr + "&key=" + apikey;
	xhr.open("GET", query, false);
	xhr.send()
	oneloaded(bar)
	var j = JSON.parse(xhr.responseText);
	return j
};

var tablerow = function(c) {
	console.log(c);
	return "<td>" + c[0] + "</td>" + "<td>"  + c[1] + "</td>" + "<td>" + c[2] + "</td>";
};

var csvrow = function(c) {
	return c[0] + "," + c[1] + "," + c[2] + "\n";
};

var initLoading = function(data) {
	elem = document.getElementById("loadingcontainer");
	bar = document.createElement("div")
	bar.className = "loadingbar";
	bar.innerText = data.length;
	bar.setAttribute("itemstoload", data.length);
	bar.setAttribute("itemsloaded", 0);
	elem.appendChild(bar);
	return bar
}

var oneloaded = function(b) {
	b.setAttribute("itemsloaded", +b.getAttribute("itemsloaded") + 1);
	b.style.width = String(b.getAttribute("itemsloaded") / b.getAttribute("itemstoload") * 100) + "%";
	b.innerText = b.getAttribute("itemsloaded");
	if(b.getAttribute("itemstoload") == b.getAttribute("itemsloaded")) {
		b.innerHTML = "<div id='bellissima' class='spin'>Bellissima!</span> " + b.innerText;
	}
}

document.getElementById("fileuri").innerText = fileuri;

d3.csv(fileuri, function(error, data) {
	if(error) {
		console.log(error);
		container.innerText = error.type;
	}
	
	console.log(data);

	bar = initLoading(data);
	
	data.forEach(function(d, i) {
		setTimeout(function() {
			georesult = resolve(d.city, bar)['results'][0]['geometry']['location'];
			entry = [d.city, georesult.lat, georesult.lng];
			coords.push(entry);
			container.innerText += csvrow(entry);

		}, i * 300);
	});
	console.log("eventually, when all timeouts have ran, use console.table(coords) to inspect what data was received");
});
